File: 백준/Silver/2606. 바이러스/바이러스.py
import sys

input = sys.stdin.readline

n = int(input())
m = int(input())

graph = [[] for _ in range(n+1)]
visited = [0 for _ in range(n+1)]

for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)



def recur(node):
    visited[node] = 1

    for nxt in graph[node]:
        if visited[nxt] == 1:
            continue
        recur(nxt)


recur(1)

print(sum(visited) - 1)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.*;
import java.util.*;

public class Main {
    static int n,m;
    static ArrayList<Integer>[] graph;
    static boolean[] visited;
    static int ans = 0;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        n = Integer.parseInt(br.readLine());
        m = Integer.parseInt(br.readLine());

        graph = new ArrayList[n+1];
        for (int i = 0; i < n + 1; i++) {
            graph[i] = new ArrayList<>();
        }
        visited = new boolean[n+1];

        for (int i = 0; i < m; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            graph[a].add(b);
            graph[b].add(a);
        }

        recur(1);
        System.out.println(ans);


    }

    static void recur(int v) {
        visited[v] = true;

        for (int node : graph[v]) {
            if (!visited[node]) {
                ans++;
                recur(node);
            }
        }
    }







}